passes/elim_dead_code.cpp

```diff
@@ -187,6 +187,20 @@ const IR::Node *ElimDeadCode::preorder(IR::MethodCallStatement *stmt) {
         }
     }
 
+    // Action annotated with `@defaultonly` is ignored before. If no action can be reachable
+    // based on previous analysis, we replace the apply call to the default action call.
+    // Only when the default action has an empty body, we remove the apply.
+    auto *defaultAction = table.getDefaultAction()->checkedTo<IR::MethodCallExpression>();
```

Comment:
Do not use `checkedTo`, it can throw an exception which can crash the server. Instead use `to` and return if the value is a nullptr.

Reply:
fixed